nucliadb_protos/knowledgebox.proto

```diff
@@ -21,6 +21,40 @@ message KnowledgeBoxID {
     string uuid = 2;
 }
 
+// External Index node provider
+enum ExternalIndexProviderType {
+    UNSET = 0;
+    PINECONE = 1;
+}
+
+// Pinecone
+message CreatePineconeConfig {
+    string api_key = 1;
+}
+
+message StoredPineconeConfig {
+    string encrypted_api_key = 1;
+
+    // Index name to index host mapping.
+    map<string, string> index_hosts = 2;
+}
+
+// External Index node provider
+message CreateExternalIndexProviderMetadata {
+    ExternalIndexProviderType type = 1;
+    oneof config {
+        CreatePineconeConfig pinecone_config = 2;
+    }
```

Comment:
No need for type if you're using `oneof`. If there's a `CreatePineconeConfig`, we know is a pinecone config, no need to store a type

Reply:
type can be useful for checking what config to parse.
Otherwise we'll need to use proto's `HasField("")` for every possibility.